### PR TITLE
Add UT for deploy.go and statefulset.go

### PR DESCRIPTION
**Pull request by realnumber666**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Promote UT coverage

**Which issue(s) this PR fixes**:
#3327 

**Does this PR introduce a user-facing change?**:
NONE

### CODE REVIEW

File: pkg/karmadactl/cmdinit/kubernetes/deploy_test.go [hunk 1]
```diff
@@ -1,8 +1,11 @@
 package kubernetes
 
 import (
+	"github.com/karmada-io/karmada/pkg/karmadactl/cmdinit/utils"
+	"net"
 	"os"
 	"testing"
+	"time"
```
Comment (RainbowMango) [inline pkg/karmadactl/cmdinit/kubernetes/deploy_test.go:8]:
```
pkg/karmadactl/cmdinit/kubernetes/deploy_test.go:4: File is not `gci`-ed with --skip-generated -s Standard,Default,Prefix(github.com/karmada-io/karmada) (gci)
	"github.com/karmada-io/karmada/pkg/karmadactl/cmdinit/utils"
pkg/karmadactl/cmdinit/kubernetes/deploy_test.go:8: File is not `gci`-ed with --skip-generated -s Standard,Default,Prefix(github.com/karmada-io/karmada) (gci)
	"time"
```

Please group the import by order `std` --> `third-party` --> `karmada`, like:

```suggestion
	"net"
	"os"
	"testing"
	"time"
	
	"github.com/karmada-io/karmada/pkg/karmadactl/cmdinit/utils"
```

**Comment by XiShanYongYe-Chang** [comment]
Hi @realnumber666, welcome to Karmada, you can sign off refer to https://github.com/karmada-io/karmada/pull/3297/checks?check_run_id=12129764078

**Comment by RainbowMango** [comment]
Thanks @realnumber666 for doing this, please sign the DCO during next push, and here is a guide line:https://github.com/karmada-io/karmada/pull/3297/checks?check_run_id=12129764078

File: pkg/karmadactl/cmdinit/kubernetes/deploy_test.go (hunk 1 above)
Comment (realnumber666) [reply pkg/karmadactl/cmdinit/kubernetes/deploy_test.go:8]: ok, I will fix it

File: pkg/karmadactl/cmdinit/kubernetes/deploy_test.go [hunk 2]
```diff
@@ -39,3 +42,232 @@ func Test_initializeDirectory(t *testing.T) {
 		})
 	}
 }
+
+func TestCommandInitOption_Validate(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (realnumber666) [reply pkg/karmadactl/cmdinit/kubernetes/deploy_test.go:46]: That seems better! I fill reorganize my tests.

File: pkg/karmadactl/cmdinit/kubernetes/deploy_test.go [hunk 3]
```diff
@@ -39,3 +43,347 @@ func Test_initializeDirectory(t *testing.T) {
...
+	tmpPath := "/tmp/pki"
+	defer os.RemoveAll(tmpPath)
+	opt := &CommandInitOption{
+		CertValidity:       365 * 24 * time.Hour,
+		EtcdReplicas:       3,
+		Namespace:          "default",
+		ExternalDNS:        "example.com",
+		ExternalIP:         "1.2.3.4",
+		KarmadaAPIServerIP: []net.IP{utils.StringToNetIP("1.2.3.5")},
+		KarmadaPkiPath:     "/tmp/pki",
```
Comment (XiShanYongYe-Chang) [inline pkg/karmadactl/cmdinit/kubernetes/deploy_test.go:156]: Here can use the `tmpPath` variable.

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Good job! Thanks~

**Review by XiShanYongYe-Chang — COMMENTED** [review]
/lgtm
/assign @lonelyCZ

**Review by RainbowMango — APPROVED** [review]
/approve

**Pull request merged**